ts/output/chtml.ts

```diff
@@ -152,31 +158,62 @@ CommonOutputJax<N, T, D, CHTMLWrapper<N, T, D>, CHTMLWrapperFactory<N, T, D>, CH
    * @override
    */
   public styleSheet(html: MathDocument<N, T, D>) {
-    if (this.chtmlStyles && !this.options.adaptiveCSS) {
+    if (this.chtmlStyles) {
+      if (this.options.adaptiveCSS) {
+        //
+        // Update the style sheet rules
+        //
+        const styles = new CssStyles();
+        this.addWrapperStyles(styles);
+        this.updateFontStyles(styles);
+        this.adaptor.insertRules(this.chtmlStyles, styles.getStyleRules());
+      }
       return this.chtmlStyles;  // stylesheet is already added to the document
     }
     const sheet = this.chtmlStyles = super.styleSheet(html);
     this.adaptor.setAttribute(sheet, 'id', CHTML.STYLESHEETID);
+    this.wrapperUsage.update();
     return sheet;
   }
 
+  /**
+   * @param {CssStyles} styles   The styles to update with newly used character styles
+   */
+  protected updateFontStyles(styles: CssStyles) {
+    styles.addStyles(this.font.updateStyles());
+  }
+
   /**
    * @override
    */
-  protected addClassStyles(CLASS: typeof CommonWrapper) {
-    if (!this.options.adaptiveCSS || (CLASS as typeof CHTMLWrapper).used) {
-      if ((CLASS as typeof CHTMLWrapper).autoStyle && CLASS.kind !== 'unknown') {
-        this.cssStyles.addStyles({
-          ['mjx-' + CLASS.kind]: {
-            display: 'inline-block',
-            'text-align': 'left'
-          }
-        });
+  protected addWrapperStyles(styles: CssStyles) {
+    if (this.options.adaptiveCSS) {
+      for (const kind of this.wrapperUsage.update()) {
+        const wrapper = this.factory.getNodeClass(kind) as any as typeof CommonWrapper;
+        wrapper && this.addClassStyles(wrapper, styles);
       }
-      super.addClassStyles(CLASS);
+    } else {
+      super.addWrapperStyles(styles);
```

Comment:
My usual: `else` to `if (!...) {... return;}`.

Reply:
Ok, ok, I'll do it.